uri.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2017 Pilosa Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilosa
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var schemeRegexp = regexp.MustCompile("^[+a-z]+$")
+var hostRegexp = regexp.MustCompile("^[0-9a-z.-]+$|^\\[[:0-9a-fA-F]+\\]$")
+var addressRegexp = regexp.MustCompile("^(([+a-z]+):\\/\\/)?([0-9a-z.-]+|\\[[:0-9a-fA-F]+\\])?(:([0-9]+))?$")
+
+// URI represents a Pilosa URI.
+// A Pilosa URI consists of three parts:
+// 1) Scheme: Protocol of the URI. Default: http.
+// 2) Host: Hostname or IP URI. Default: localhost. IPv6 addresses should be written in brackets, e.g., `[fd42:4201:f86b:7e09:216:3eff:fefa:ed80]`.
+// 3) Port: Port of the URI. Default: 10101.
+//
+// All parts of the URI are optional. The following are equivalent:
+// 	http://localhost:10101
+// 	http://localhost
+// 	http://:10101
+// 	localhost:10101
+// 	localhost
+// 	:10101
+type URI struct {
+	scheme string
+	host   string
+	port   uint16
+	error  error
+}
+
+// DefaultURI creates and returns the default URI.
+func DefaultURI() *URI {
+	return &URI{
+		scheme: "http",
+		host:   "localhost",
+		port:   10101,
+	}
+}
+
+// NewURIFromHostPort returns a URI with specified host and port.
+func NewURIFromHostPort(host string, port uint16) (*URI, error) {
+	uri := DefaultURI()
+	err := uri.SetHost(host)
+	if err != nil {
+		return nil, err
+	}
+	uri.SetPort(port)
+	return uri, nil
+}
+
+// NewURIFromAddress parses the passed address and returns a URI.
+func NewURIFromAddress(address string) (*URI, error) {
+	uri, err := parseAddress(address)
+	if err != nil {
+		return &URI{error: err}, err
+	}
+	return uri, err
+}
+
+// URIFromAddress creates a URI from the given address.
+func URIFromAddress(host string) *URI {
```

Comment:
Why does this function exist?

Reply:
It is for optionally ignoring errors while creating URIs to make the code a bit less verbose when the address is most probably valid. It sets the `error` field instead of returning it, so it can be used in function calls. It's not being used, so removed that, `error` and related stuff